app/controllers/incidents_controller.rb

```diff
@@ -10,9 +10,15 @@ class IncidentsController < ApplicationController
 
   def batch_hastus_export
     @incidents = Incident.where(id: params[:ids])
-    send_data render_to_string('batch_export.xml.haml'),
-      filename: "#{@incidents.map(&:id).sort.join(',')}.xml",
-      disposition: 'attachment'
+    if params[:format] == 'xml-button'
```

Comment:
I should probably change it so that the format only includes the format portion. No button.

Reply:
Perhaps, although isn't format also a separate attribute of the request that you can use as well?

`respond_to do |format|` and all that?